docs/creating-date-objects-for-chart-data.md

```diff
@@ -0,0 +1,80 @@
+# Creating Date Objects For Chart Data
+
+> There are many ways to format a date as a string. The JavaScript specification only specifies one format to be universally supported: the [date time string format](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-date-time-string-format), a simplification of the ISO 8601 calendar date extended format. The format is as follows:
+> ```
+> YYYY-MM-DDTHH:mm:ss.sssZ
+> ```
+
+Various components can be omitted from the above format, and other formats can also be used to create a date object, resulting in the following types of interpretations:
+
+### UTC time
+
+The ISO format is interpreted as UTC time in the following forms. Notice how the components in the input string match those in the output ISO string.
+
+1. Date-only form: `YYYY`, `YYYY-MM`, `YYYY-MM-DD`
+
+   ```js
+   console.log(new Date('2010').toISOString())
+   // Output: "2010-01-01T00:00:00.000Z"
+
+   console.log(new Date('2010-10').toISOString())
+   // Output: "2010-10-01T00:00:00.000Z"
+
+   console.log(new Date('2010-10-10').toISOString())
+   // Output: "2010-10-10T00:00:00.000Z"
+   ```
+
+2. Date-time form (with **Z**ero time zone offset): One of the above date-only forms, followed by `T`, followed by `HH:mm`, `HH:mm:ss`, or `HH:mm:ss.sss`. Each combination followed by the literal character `Z`.
+
+   ```js
+   console.log(new Date('2010-10-10T02:10Z').toISOString())
+   // Output: "2010-10-10T02:10:00.000Z"
+
+   console.log(new Date('2010-10-10T02:10:10Z').toISOString())
+   // Output: "2010-10-10T02:10:10.000Z"
+
+   console.log(new Date('2010-10-10T02:10:10.100Z').toISOString())
+   // Output: "2010-10-10T02:10:10.100Z"
+   ```
+
+There are alternative methods to create a date object that will be interpreted as UTC time, such as using the [Date.UTC()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/UTC) static method or directly passing the time in milliseconds that has elapsed since the [epoch](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-time-values-and-time-range) as an argument to the [Date() constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date).
+
+```js
+console.log(new Date(Date.UTC(2010, 9, 10, 2, 10)).toISOString())
+// Output: "2010-10-10T02:10:00.000Z"
+
+console.log(new Date(1286676600000).toISOString())
+// Output: "2010-10-10T02:10:00.000Z"
+```
+
+### Local time
+
+All other formats (including the above date-time form of the ISO format when the time zone offset is absent) are interpreted as local time. Notice how the components in the input string don't completely match those in the output ISO string.
+
+```js
+console.log(new Date('2010-10-10T02:10').toISOString())
+// Output (IST): "2010-10-09T20:40:00.000Z"
```

Comment:
provide a couple of examples for other time zones as well for completeness.

Reply:
done